openresty-build-tools/kong-ngx-build

```diff
@@ -597,20 +597,33 @@ main() {
 
         OPENRESTY_RPATH=${OPENRESTY_RPATH:-$OPENSSL_INSTALL/lib}
 
-        if ld --disable-new-dtags 2>&1 >/dev/null | grep -q "disable-new-dtags"; then
-          OPENRESTY_OPTS+=("--with-ld-opt='-L$OPENSSL_INSTALL/lib -Wl,-rpath,$OPENRESTY_RPATH'")
+
+        if [ -n "$BORINGSSL_VER" ]; then
+          OPENRESTY_OPTS+=(
+            "--with-cc-opt='-I$BORINGSSL_DOWNLOAD/.openssl/include'"
+            "--with-ld-opt='-Wl,-Bsymbolic-functions -Wl,-z,relro -L$BORINGSSL_DOWNLOAD/.openssl/lib -Wl,--disable-new-dtags,-rpath,$OPENRESTY_RPATH'"
+          )
+
         else
-          OPENRESTY_OPTS+=("--with-ld-opt='-L$OPENSSL_INSTALL/lib -Wl,--disable-new-dtags,-rpath,$OPENRESTY_RPATH'")
-        fi
+          if ld --disable-new-dtags 2>&1 >/dev/null | grep -q "disable-new-dtags"; then
+            OPENRESTY_OPTS+=("--with-ld-opt='-L$OPENSSL_INSTALL/lib -Wl,-rpath,$OPENRESTY_RPATH'")
+          else
+            OPENRESTY_OPTS+=("--with-ld-opt='-L$OPENSSL_INSTALL/lib -Wl,--disable-new-dtags,-rpath,$OPENRESTY_RPATH'")
+          fi
 
-        OPENRESTY_OPTS+=("--with-cc-opt='-I$OPENSSL_INSTALL/include'")
+          OPENRESTY_OPTS+=("--with-cc-opt='-I$OPENSSL_INSTALL/include'")
+        fi
 
         if [ $DEBUG == 1 ]; then
           OPENRESTY_OPTS+=('--with-debug')
           OPENRESTY_OPTS+=('--with-luajit-xcflags="-DLUAJIT_USE_VALGRIND -DLUA_USE_ASSERT -DLUA_USE_APICHECK -DLUAJIT_USE_SYSMALLOC"')
         fi
 
         eval ./configure ${OPENRESTY_OPTS[*]}
+
+        if [ -n "$BORINGSSL_VER" ]; then
+          touch $BORINGSSL_DOWNLOAD/.openssl/include/openssl/ssl.h
```

Comment:
Just being curious, what does this line do? Create an empty file or just touch the timestamp? What does it solve?

Reply:
It updates the timestamp. When I left it out, several symbols were left undefined. I think it's because of the meeting between cmake+ninja on one side and configure+make on the other.